src/crypto/tls/delegated_credentials.go

```diff
@@ -0,0 +1,493 @@
+// Copyright 2018 The Go Authors. All rights reserved.
```

Comment:
There might be external dependencies that need to generate and marshal DCs. I wonder if it makes sense to move this file into its own package, e.g.., crypto/tls/subcerts or crypto/tls/delegated_credentials?

This is the plan for the ECH config minting logic: see #29.

Reply:
Mmm.. very interesting. It might be the best.

Reply:
On second thought, I'm not so sure my suggestion is a good idea. I'm finding that the ECH minting logic needs to know about TLS alerts:
https://github.com/cloudflare/go/pull/29/files#diff-7de0736854ce15367fc2dd7dd7c21d97R487

These aren't in the public API for crypto/tls, so it seems that I need to move crypto/tls/ech into crypto/tls.

Reply:
I would advise against a subpackage, this is an internal implementation detail that should remain private. Most details should probably not be exposed through a public API. If you look at the current crypto/tls implementation, there is currently no subpackages, so there must have been a good reason for that.

Some other projects end up forking the standard library or copying details over if they need to share functionality.

Reply:
I agree @Lekensteyn and @cjpatton